logsrvd/logsrvd.c

```diff
@@ -80,6 +80,9 @@ static const char server_id[] = "Sudo Audit Server 0.1";
 static const char *conf_file = _PATH_SUDO_LOGSRVD_CONF;
 static double random_drop;
 
+/* event timeout for read / write / tls handshake */
+static struct timespec timeo = {0, 0};
```

Comment:
A timeout of zero is not the same as no timeout.  A zero timeout will return immediately whereas a NULL timespec pointer means an infinite timeout.  You might consider making logsrvd_conf_get_sock_timeout() return a struct timespec * and store the struct timespec in struct logsrvd_config_server instead of an int.  Then  logsrvd_conf_get_sock_timeout() could check sudo_timespecisset(&logsrvd_config->server.timeout) and return NULL if not set, else return the timespec pointer.

Reply:
This is just an initialization to zero after declaration. `timeo` is set in the `main()` from the config, but you are right, it is better to store the timeout as `struct timespec` rather than as `int`.

I fixed this, however setting timeout to zero in the configuration or not set timeout value at all will eventually be the same for `sudo_timespecisset()`, therefore `logsrvd_conf_get_sock_timeout()` will return NULL in both cases which means infinite timeout.

In my opinion, zero timeout wouldn't make any sense anyway, each client connection attempt would be timeout at the TLS handshake phase.